redbeat/schedulers.py

```diff
@@ -159,15 +159,24 @@ def schedule(self, value):
         else:
             self.app.conf.CELERYBEAT_SCHEDULE = value
 
-    def either_or(self, name, default=None):
+    def warn_if_wrong_format(self, name):
         if CELERY_4_OR_GREATER and name == name.upper():
             warnings.warn(
                 'Celery v4 installed, but detected Celery v3 '
                 'configuration %s (use %s instead).' % (name, name.lower()),
                 UserWarning
             )
+    def key_has_value_or(self, name, default=None):
+        self.warn_if_wrong_format(name)
         return self.app.conf.first(name, name.upper()) or default
 
+    def key_present_or(self, name, default=None):
```

Comment:
Should there be tests for this function?

Reply:
Definitely.  I'll add one.